third_party/move/move-compiler-v2/src/file_format_generator/mod.rs

```diff
@@ -106,6 +121,12 @@ pub fn generate_file_format(
             }));
         }
     }
+    for (id, (m, map)) in module_data {
+        env.attach_compiled_module(id, m, map)
```

Comment:
This really should not happen here: in the normal code path, we do not need to attach bytecode. It also duplicates logic which is already here: https://github.com/aptos-labs/aptos-core/blob/d6483636c325f76c5207bb20f6160f0f395cba39/third_party/move/move-compiler-v2/src/lib.rs#L142

Reply:
Added a experiment flag to control when to attach compiled modules.